protected/humhub/events/MigrationEvent.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * @link      https://www.humhub.org/
+ * @copyright Copyright (c) 2023 HumHub GmbH & Co. KG
+ * @license   https://www.humhub.com/licences
+ */
+
+namespace humhub\events;
+
+use humhub\components\Event;
+use humhub\interfaces\ApplicationInterface;
+use yii\base\Module;
+
+/**
+ * @property int|null $result Result of the migration:
+ * ``
+ *  - `ExitCode::OK`: Success;
+ *  - `ExitCode::UNSPECIFIED_ERROR`: failure;
+ *  - `Null`: nothing done
+ * ``
+ */
+class MigrationEvent extends Event
+{
+    /**
+     * @var \humhub\components\Module|ApplicationInterface|null
+     */
+    public ?Module $module;
+
+    /**
+     * @var string|null path to the migrations folder for modules, empty for application
+     */
+    public ?string $path;
+
+    /**
+     * @var string Either `up` or `uninstall`
+     */
+    public string $migration;
+
+    /**
+     * @var string|null Error message to be used in the log
+     */
+    public ?string $errorMessage = null;
```

Comment:
When `errorMessage` null, means no error?

Reply:
No, when set to `null` the [standard message](https://github.com/humhub/humhub/pull/6550/files/1b50ea92d6a9f8f534abe34389372de1220b4710#diff-381c812c100b93c52f71ee2cd25c98821a7436b5d1223f2f2d96059fcc05950dR237) will be used. This allows a module developper to add custom error message. But this can be removed or renamed, if preferred.

Reply:
I would prefer to remove. Should be no need to maniplulate the error message.